File: LinkedLists/DeletingNode/java/code.java
class LinkedList {

	Node head;

	public class Node {
		Node next;
		int data;
		Node(int val) {
			next=null;
			data=val;
		}
	}

	public void push(int newData) {
		Node newNode = new Node(newData);
		if(head==null) {
			head=newNode;
			return;
		}
		newNode.next = head;
		head = newNode;
		return;
	}

	public void printList() {
		Node temp = head;
		while(temp!=null) {
			System.out.print(temp.data+" ");
			temp=temp.next;
		}
	}

	public void deleteNode(int key) {
		Node temp=head;
		Node prev=null;

		if(temp!=null && temp.data==key) {
			head=temp.next;
			return;
		}

		while(temp.data!=key) {
			prev=temp;
			temp=temp.next;
		}

		prev.next=temp.next;

	}

	public static void main(String[] args) {

		LinkedList list = new LinkedList();

		list.push(1);
		list.push(2);
		list.push(3);
		list.push(4);

		list.printList();

		list.deleteNode(2);

		list.printList();
		
	}

}

File: LinkedLists/RemoveDuplicateSortedLL/code.cpp
#include <bits/stdc++.h>
using namespace std;

class Node {
public:
	Node* next;
	int data;
	Node(int val) {
		next=NULL;
		data=val;
	}
};

void push(Node** headRef, int newData) {
	Node* newNode = new Node(newData);
	if(*headRef==NULL) {
		*headRef=newNode;
		return;
	}
	newNode->next=*headRef;
	*headRef=newNode;
	return;
}

void printList(Node* head) {
	while(head!=NULL) {
		cout << head->data << " ";
		head=head->next;
	}
	cout << endl;
	return;
}

void removeDuplicate(Node* head) {
	if(head==NULL) {
		return;
	}
	Node* temp=head;
	while(temp->next!=NULL) {
		if(temp->data == temp->next->data) {
			Node* tmp = temp->next->next;
			free(temp->next);
			temp->next = tmp;
		}
		else {
			temp=temp->next;	
		}		
	}
}

int main() {

	Node* head=NULL;
	push(&head, 1);
	push(&head, 1);
	push(&head, 2);
	push(&head, 2);
	push(&head, 3);
	push(&head, 4);

	printList(head);

	removeDuplicate(head);

	printList(head);

	return 0;
}